starter-code/src/clue.js

```diff
@@ -1,94 +1,259 @@
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+const diningRoom ={
+    name: "Dining Room"
+}
+const conservatory ={
+    name: "Conservatory"
+}
+const kitchen ={
+    name: "Kitchen"
+}
+const study ={
+    name: "Study"
+}
+const library ={
+    name: "Library"
+}
+const billiardRoom ={
+    name: "Billiard Room"
+}
+const lounge ={
+    name: "Lounge"
+}
+const ballroom ={
+    name: "Ballroom"
+}
+const hall ={
+    name: "Hall"
+}
+const spa ={
+    name: "Spa"
+}
+const livingRoom ={
+    name: "Living Room"
+}
+const observatory ={
+    name: "Observatory"
+}
+const theater ={
+    name: "Theater"
+}
+const guestHouse ={
+    name: "Guest House"
+}
+const patio ={
+    name: "Patio"
+}
 
 // Characters Collection
 const charactersArray = [];
 
+charactersArray.push(mrGreen);
+charactersArray.push(drOrchid);
+charactersArray.push(profPlum);
+charactersArray.push(missScarlet);
+charactersArray.push(mrsPeacock);
+charactersArray.push(mrMustard);
+
 // Rooms Collection
 const roomsArray = [];
 
+roomsArray.push(diningRoom);
+roomsArray.push(conservatory);
+roomsArray.push(kitchen);
+roomsArray.push(study);
+roomsArray.push(library);
+roomsArray.push(billiardRoom);
+roomsArray.push(lounge);
+roomsArray.push(ballroom);
+roomsArray.push(hall);
+roomsArray.push(spa);
+roomsArray.push(livingRoom);
+roomsArray.push(observatory);
+roomsArray.push(theater);
+roomsArray.push(guestHouse);
+roomsArray.push(patio);
+
 // Weapons Collection
 const weaponsArray = [];
+
+weaponsArray.push(rope);
+weaponsArray.push(knife);
+weaponsArray.push(candlestick);
+weaponsArray.push(dumbbell);
+weaponsArray.push(poison);
+weaponsArray.push(axe);
+weaponsArray.push(bat);
+weaponsArray.push(trophy);
+weaponsArray.push(pistol);
+
+///////////////////////////////////////
+let mysteryEnvelope = {
+}
+
+function randomSelector(array) {
+    const length = array.length;
+    if (length === 0) return undefined;
+
+    const random = Math.floor( Math.random() * length);
+
+    return array[random];
+}
+
+function pickMystery (){
+    let suspect = randomSelector(charactersArray);
+    let weapon = randomSelector(weaponsArray);
+    let room = randomSelector(roomsArray);
+
+    let mysteryEnvelope = { };
+    if( suspect !== undefined){
+        mysteryEnvelope.suspect = suspect;
+    }
+    if( weapon !== undefined){
+        mysteryEnvelope.weapon = weapon;
+    }
+    if( room !== undefined){
+        mysteryEnvelope.room = room;
+    }
+
+    return mysteryEnvelope;
+}
+
+function revealMystery(mistery){
+    let result = "";
+    
+    if (mistery.suspect === undefined || (mistery.suspect.first_name === undefined && mistery.suspect.last_name === undefined) ){
+        result += "Unknow suspect";
+    } else{
+        if (mistery.suspect.last_name === undefined){
+            result += mistery.suspect.first_name;
+        } else if (mistery.suspect.first_name === undefined){
```

Comment:
Por qué evaluáis cada caso `undefined` e igualáis el caso contrario? no sería mejor que cada condición evalúa si es `truthy` y entonces lo guardáis? Haciéndolo como lo tenéis puede crear confusión.

Y recordad negar `truthy` y evitar condiciones anidadas como la que tenéis. Anidar demasiadas condiciones es `code smell`